tools/pd-ctl/pdctl/command/cluster_command.go

```diff
@@ -19,10 +19,10 @@ import "github.com/spf13/cobra"
 // NewClusterCommand return a cluster subcommand of rootCmd
 func NewClusterCommand() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:               "cluster",
-		Short:             "show the cluster information",
-		PersistentPreRunE: requirePDClient,
-		Run:               showClusterCommandFunc,
+		Use:     "cluster",
+		Short:   "show the cluster information",
+		PreRunE: requirePDClient,
```

Comment:
https://github.com/tikv/pd/blob/199b01792159e5d8e83ef419a5053401e998bb0e/tools/pd-ctl/pdctl/ctl.go#L40

`RequireHTTPSClient` should be called firstly

Reply:
PersistentPreRunE is called before PreRunE?

Reply:
Why change to `PreRunE`? Using `PersistentPreRunE` ensures that all sub-commands can inherit this requirement without setting `PreRunE` for each.

Reply:
> PersistentPreRunE is called before PreRunE?

Yes.

> Why change to PreRunE? Using PersistentPreRunE ensures that all sub-commands can inherit this requirement without setting PreRunE for each.

For this purpose, how about we put `command.RequireHTTPSClient` into function `requirePDClient`? @JmPotato

Reply:
LGTM.

Reply:
After [EnableTraverseRunHooks](https://github.com/tikv/pd/pull/8239/commits/44688fc5c17554d8e9c568bccacab362816d6f1c), the execution order of `PersistentPreRunE` is `rootCmd -> subCmd`, so the problem has been solved.